apps/src/blockly/addons/cdoFieldImageDropdown.js

```diff
@@ -1,9 +1,36 @@
-import CdoFieldDropdown from './cdoFieldDropdown';
+import {FieldGridDropdown} from '@blockly/field-grid-dropdown';
+import color from '@cdo/apps/util/color';
 
-export class CdoFieldImageDropdown extends CdoFieldDropdown {
+// Note that this class *does not* inherit from CdoFieldDropdown
+export class CdoFieldImageDropdown extends FieldGridDropdown {
   constructor(menuGenerator, width, height, buttons) {
-    const fixedMenuGenerator = fixMenuGenerator(menuGenerator, width, height);
-    super(fixedMenuGenerator);
+    // We have to decide how many columns to have when we create the block. The
+    // number of options in the block can change over time, but we can just use
+    // the initial number of options to determine how many columns to show- we're
+    // assuming that the number of options usually won't change that drastically,
+    // so the number of columns can probably stay the same.
+    const initialOptions = fixMenuGenerator(menuGenerator, width, height);
+    const numColumns =
+      initialOptions.length <= 7
+        ? 1
+        : Math.floor(Math.sqrt(initialOptions.length));
```

Comment:
Hah, so we're deciding we generally want an equal number of rows and columns?

Reply:
Yeah, I pulled this logic from https://github.com/code-dot-org/blockly/blob/main/core/ui/fields/field_rectangular_dropdown.js#L315
We can easily change it if we want later, but for now I'm going for feature-parity with old blockly